core/src/main/java/org/elasticsearch/action/admin/indices/shards/IndicesShardsStoresResponse.java

```diff
@@ -0,0 +1,421 @@
+                        return REPLICA;
+                    case 2:
+                        return UNUSED;
+                    default:
+                        throw new IllegalArgumentException("unknown id for allocation [" + id + "]");
+                }
+            }
+
+            private static Allocation readFrom(StreamInput in) throws IOException {
+                return fromId(in.readByte());
+            }
+
+            private void writeTo(StreamOutput out) throws IOException {
+                out.writeByte(id);
+            }
+        }
+
+        private ShardStoreStatus() {
+        }
+
+        ShardStoreStatus(DiscoveryNode node, long version, Allocation allocation, Throwable storeException) {
+            this.node = node;
+            this.version = version;
+            this.allocation = allocation;
+            this.storeException = storeException;
+        }
+
+        /**
+         * Node the shard belongs to
+         */
+        public DiscoveryNode getNode() {
+            return node;
+        }
+
+        /**
+         * Version of the shard, used to select the store that will be
+         * used as the very first primary.
+         */
+        public long getVersion() {
+            return version;
+        }
+
+        /**
+         * Exception while trying to open the
+         * shard index or from when the shard failed
+         */
+        public Throwable getStoreException() {
+            return storeException;
+        }
+
+        /**
+         * The allocation status of the store
+         * @see Allocation
+         */
+        public Allocation getAllocation() {
+            return allocation;
+        }
+
+        static ShardStoreStatus readShardStoreStatus(StreamInput in) throws IOException {
+            ShardStoreStatus shardStoreStatus = new ShardStoreStatus();
+            shardStoreStatus.readFrom(in);
+            return shardStoreStatus;
+        }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            node = DiscoveryNode.readNode(in);
+            version = in.readLong();
+            allocation = Allocation.readFrom(in);
+            if (in.readBoolean()) {
+                storeException = in.readThrowable();
+            }
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            node.writeTo(out);
+            out.writeLong(version);
+            allocation.writeTo(out);
+            if (storeException != null) {
+                out.writeBoolean(true);
+                out.writeThrowable(storeException);
+            } else {
+                out.writeBoolean(false);
+            }
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            node.toXContent(builder, params);
+            builder.field(Fields.VERSION, version);
+            builder.field(Fields.ALLOCATED, allocation.name());
```

Comment:
Added a `Allocation#value()` method for the string representation instead of using `name()`